lib/engine/action/buy_train.rb

```diff
@@ -5,14 +5,15 @@
 module Engine
   module Action
     class BuyTrain < Base
-      attr_reader :train, :price, :exchange
+      attr_reader :train, :price, :exchange, :ability
 
-      def initialize(entity, train:, price:, variant: nil, exchange: nil)
+      def initialize(entity, train:, price:, variant: nil, exchange: nil, ability: nil)
```

Comment:
we should not do this (add ability here)

instead, just change the price when you buy the train

Reply:
Fixed